[Inductor] Add NaN assert to returned values from generated code

PaulZhang12 opened the pull request:
Summary: It is possible to have `reinterpret_tensor` in the output of inductor codegen, e.g. `reinterpret_tensor(buf366, (1024, ), (1, ), 0)` in the return tuple. This adds assertions to all return values from inductor codegen to prevent nans from slipping through and being hard to trace.

Test Plan:
NaN asserts properly generated in example gemm script:

    vars = (buf1, primals_2, buf2, primals_1, )
    for var in vars:
        if isinstance(var, torch.Tensor):
            assert not var.isnan().any().item()
            assert not var.isinf().any().item()




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

eellison reviewed (commented):
Could we add this to https://github.com/pytorch/pytorch/blob/6f992e1b3f113126246dbd10facaae82492c5534/torch/_inductor/codegen/wrapper.py#L1731-L1758 instead ? 

ReinterpretView is also used on intermediaries etc.

eellison commented:
but need to fix test failures

PaulZhang12 committed:
[Inductor] Add NaN assert to returned values from generated code (#154455)

Summary:

It is possible to have `reinterpret_tensor` in the output of inductor codegen, e.g. `reinterpret_tensor(buf366, (1024, ), (1, ), 0)` in the return tuple. This adds assertions to all return values from inductor codegen to prevent nans from slipping through and being hard to trace.

Test Plan:
NaN asserts properly generated in example gemm script:

    vars = (buf1, primals_2, buf2, primals_1, )
    for var in vars:
        if isinstance(var, torch.Tensor):
            assert not var.isnan().any().item()
            assert not var.isinf().any().item()

Differential Revision: D74691131

huydhn commented:
@pytorchbot revert -m 'Sorry for reverting your change, I am trying to see if it help fix the broken trunk below.  It it does not help, I will reland the PR' -c weird

inductor/test_compile_subprocess.py::CpuTests::test_AllenaiLongformerBase_repro_cpu [GH job link](https://github.com/pytorch/pytorch/actions/runs/15338318721/job/43159787447) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/c3de2c7c6bc865b9fabd2db8f2af6383936aa653)

huydhn commented:
@pytorchbot merge -f 'Reland the same code'

huydhn commented:
@pytorchbot merge -f 'Reland the same code'
